### PR TITLE
Support pcm drain ops

### PR DESCRIPTION
**Pull request by quic-guodhu**
Support pcm_drain in pcm_hw and pcm_plugin.

### CODE REVIEW

File: include/tinyalsa/plugin.h [hunk 1]
```diff
@@ -124,7 +124,10 @@ struct pcm_plugin_ops {
     int (*prepare) (struct pcm_plugin *plugin);
     /** Start data transfer from/to the plugin */
     int (*start) (struct pcm_plugin *plugin);
-    /** Drop pcm frames */
+    /** Signal the plugin to drain PCM */
+    int (*drain) (struct pcm_plugin *plugin);
+    /** Stop a PCM dropping pending frames if drian() is NOT called.
```
Comment (rohit-qc) [inline include/tinyalsa/plugin.h:129]: fix drain misspell

File: src/pcm_plugin.c [hunk 2]
```diff
@@ -67,6 +67,7 @@ enum {
     PCM_PLUG_STATE_SETUP,
     PCM_PLUG_STATE_PREPARED,
     PCM_PLUG_STATE_RUNNING,
+    PCM_PLUG_STATE_DRAINING,
```
Comment (rohit-qc) [inline src/pcm_plugin.c:70]: what is the purpose of this state

Comment (quic-guodhu) [reply src/pcm_plugin.c:70]: Currently, this is not used anywhere. But STATE_DRAINING is defined at pcm: https://github.com/tinyalsa/tinyalsa/blob/4fbaeef03cd1cb216e0f356c0433ca70f8b9c464/include/tinyalsa/pcm.h#L127
I thought it's more complete to define this in plugin as well.
What's your advice? Should I remove this?

Comment (rohit-qc) [reply src/pcm_plugin.c:70]: I don't think there is a need to add this state unless we are using it for handling some scenarios

Comment (quic-guodhu) [reply src/pcm_plugin.c]: Thanks. Accept and update.

File: src/pcm.c [hunk 3]
```diff
@@ -1218,6 +1218,22 @@ int pcm_start(struct pcm *pcm)
...
+ * @param pcm A PCM handle.
+ * @return On success, zero; on failure, a negative number.
+ * @ingroup libtinyalsa-pcm
+ */
+int pcm_drain(struct pcm *pcm)
+{
+    if (!pcm_is_ready(pcm))
+        return -1;
+
+    if (pcm->ops->drain(pcm->data) < 0)
```
Comment (dvdli) [inline src/pcm.c:1231]: Why don't you use `pcm->ops->ioctl(pcm->data, SNDRV_PCM_IOCTL_DRAIN)`, but add a new function?

Comment (quic-guodhu) [reply src/pcm.c]: Thanks for the comment. I have updated to simplify the logic. @rohit-qc @dvdli can you help review again.

**Review by dvdli — APPROVED** [review]
LGTM

**Review by rohit-qc — APPROVED** [review]

**Pull request merged**